### PR TITLE
added a first run of RU locale, added some POS in the definitions output

### PR DESCRIPTION
**Pull request by victornove**
added RU locale, only few templates though as had memory issues with find-templates
modified some DE (flattened definitions with ordered lists) - can remove if needed?
modified definitions to include POS (checked for EN, FR, DE, RU) - had to modify DE to pull POS from different section into definition. - can also remove if needed?

### CODE REVIEW

**Comment by BoboTiG** [comment]
That's awesome!

I would be more comfortable with several PRs to target specific changes. When you modify existent code, it will need tests :).
Let's tackle only the new locale here.

Also, I am not sure about other changes, mostly because I do not see a before/after visual, it may help a lot here. Let's open new PRs and discuss about it there.

:+1:

**Comment by BoboTiG** [comment]
> I'm entirely clueless about the testing part (I can read through some docs if needed?). Otherwise , if I manage to get the list of templates (find-templates gets killed by my machine or throws memory errors) and start working through them, I can add more to the template handler.

Testing is easy when you've done it once ;)

Let's say you want to test a word (I cannot propose something, I really do not read Russian), picd that work on the Wiktionary, and from the "edit-mode" tab, copy the Wikicode. Then, in the folder `tests/data/ru/` create a file named `<word>.wiki`, and poste the copied Wikicode inside it.
Now, you have the Wiktionary word file, let's add the real test. You could copy any other test file, you seems to work in France, so maybe are you fluent in French, and it will be easier to copy-paste `tests/test_fr.py` as `tests/test_ru.py`. Anyway, here is the minimal skeleton for such test file:

```python
import pytest

from wikidict.render import parse_word
from wikidict.utils import process_templates


@pytest.mark.parametrize(
    "word, pronunciations, gender, etymology, definitions",
    [
        # Each and every word must contain all those details, it can be empty list or string when the information is not available
        (
            "<word>",  # this must be the same word (case-sensitivity is important) as "tests/data/ru/<word>.wiki"
            ["pronunciation1", ...],
            "gender",
            ["etymology"],
            ["definition1", ...],
        ),
    ],
)
def test_parse_word(
    word, pronunciations, gender, etymology, definitions, page
):
    """Test the sections finder and definitions getter."""
    code = page(word, "ru")
    details = parse_word(word, code, "ru", force=True)
    assert pronunciations == details.pronunciations
    assert gender == details.gender
    assert definitions == details.definitions
    assert etymology == details.etymology
```
This is a complete test for a given word. You can add as many words as files you will create in `tests/data/ru/*.wiki`. Those tests are the most important to validate the parsing of Russian words. By default, you should add words that cover parsed sections. If you find one word covering all sections, that's great too!

Then, maybe not now, but when you will work on adding templates, you will have to create lighter tests to only cover the template transformation. Here is the skeleton:

```python
@pytest.mark.parametrize(
    "wikicode, expected",
    [
        # ("minimal wikicode template", "expected parsed result")
        ("{{1|Descendant}}", "Descendant"),
        ("{{1er}}", "1<sup>er</sup>"),
        ("{{1er|mai}}", "1<sup>er</sup>&nbsp;mai"),
    ],
)
def test_process_templates(wikicode, expected):
    """Test templates handling."""
    assert process_templates("foo", wikicode, "ru") == expected
```

---

About `find-templates` issues, if you can manage to open an issue with the command, and the error(s), it would be great.

**Comment by BoboTiG** [comment]
Could you open an issue about the `find-templates` error (with the exact command, and the error) so that we could investigate.

> PS: for the visuals, you mean visualising the difference output? Something like this?

Yes, exactly :)

> Also, I was wondering if sometimes using the expanded templates via '&templates=expand' isn't easier? for example I can't find a way to get the pronunciation in russian without it.

@lasconic do you have a opinion?

**Comment by lasconic** [comment]
>> Also, I was wondering if sometimes using the expanded templates via '&templates=expand' isn't easier? for example I can't find a way to get the pronunciation in russian without it.

> @lasconic do you have a opinion?

If I understand correctly, this is not an option, since it would work only with `get-word` and not with the whole dump where templates are not "expanded". Right ?

**Comment by lasconic** [comment]
Regarding POS, let's discuss it somewhere else. Maybe here : https://github.com/BoboTiG/ebook-reader-dict/issues/1149

File: wikidict/render.py [hunk 1]
```diff
@@ -268,7 +274,7 @@ def find_all_sections(code: str, locale: str) -> List[Tuple[str, wtp.Section]]:
                 )
             )
 
-    def section_title(title: str) -> str:
+    def section_title(title: str, locale='de') -> str:
```
Comment (lasconic) [inline wikidict/render.py:277]: I believe it would be better to add "ru" to the list instead ?

Comment (victornove) [reply wikidict/render.py]: I can't remember what I was doing here, I think it is just an error on my part. I removed it.

**Comment by victornove** [comment]
The problem with find-templates was simply my linux machine running out of memory. I split the json file and worked through it by pieces... not sure its worth opening a ticket for that? Turns out because of [this issue](https://github.com/BoboTiG/ebook-reader-dict/issues/1275) there are 18000 templates to work through though!  

Also, I'm kind of struggling with the details of getting my tests working, but I think(pray) this last correction will work! Once I get the first test working I can add some more.

**Review by BoboTiG — COMMENTED** [review]
To fix quality code issues, just run `./check.sh` locally, and push changes.

**Comment by BoboTiG** [comment]
> (is there a way to run unit tests selectively? like only for russian language/changed files?)

`python -m pytest tests/test_ru.py`

**Comment by BoboTiG** [comment]
@all-contributors please add @victornove  for code

**Comment by BoboTiG** [comment]
Thanks a lot @victornove :champagne:

**Pull request merged**